.github/workflows/build-linux.yml

```diff
@@ -0,0 +1,214 @@
+name: Build Linux
+
+on:
+  workflow_call:
+    inputs:
+      binary_list:
+        required: true
+        type: string
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
```

Comment:
I don't think we should use the latest image here, because it effectively determines the minimum version of glibc required by the binaries. E.g. I've checked [your test binaries](https://github.com/erubboli/mintlayer-core-test-build/releases/tag/v0.99.15) - according to `objdump`, `node-daemon` needs at least one symbol from glibc v2.34, while `node-gui` needs a symbol from v2.35.
Glibc v2.35 is the one used by Ubuntu 22.04, which is what "ubuntu-latest" [currently means on GitHub Actions](https://github.com/actions/runner-images).  But they'll update to 24.04 eventually and our next release may suddenly require even a newer version of a Linux distro.
For comparison, Debian 11 (which, as I've googled, will be supported until 08.2026), has glibc v2.31, so the binaries won't work there.
And RHEL 8 (supported until 2029) has glibc v2.28.

So at the very least I'd suggest to specify a concrete Linux distro here.
GitHub actions should still support ubuntu-20.04 (which uses glibc v2.31), we can use this one, so that the binaries work on Debian 11 at least.

An alternative solution might be to use one of the "manylinux" docker images, which are used to compile Python extensions. I've found [this article](https://kobzol.github.io/rust/ci/2021/05/07/building-rust-binaries-in-ci-that-work-with-older-glibc.html) where the author has successfully used one of the manylinux images with GitHub actions. 

---
Or we may want to switch to using musl-libc targets, so that the binaries don't depend on glibc at all. Though those targets are still considered ["Tier 2"](https://doc.rust-lang.org/nightly/rustc/platform-support.html#tier-2-with-host-tools) so I'm not sure if it's a good idea (in any case, it's not a good idea to switch to musl in the current release).

P.S. We could also provide musl-based binaries in addition to glibc-based ones, so that people on older systems could have an alternative.